src/backend/base/langflow/components/models/embedding_model.py

```
@@ -77,37 +78,53 @@
-        chunk_size = self.chunk_size
-        request_timeout = self.request_timeout
-        max_retries = self.max_retries
-        show_progress_bar = self.show_progress_bar
-        model_kwargs = self.model_kwargs or {}
-
         if provider == "OpenAI":
-            if not api_key:
-                msg = "OpenAI API key is required when using OpenAI provider"
-                raise ValueError(msg)
+            if not self.api_key:
+                raise ValueError("OpenAI API key is required for OpenAI provider.")
```

Ruff (0.11.9)

Avoid specifying long messages outside the exception class

(TRY003)